### PR TITLE
RW-9840 Add SAS helm chart

### PR DESCRIPTION
**Pull request by yonghaoy**
https://precisionmedicineinitiative.atlassian.net/browse/RW-9840

Pred Steps:

1. Create personal Google project with GCR and GKE enabled.
2. Create GKE cluster, setup workload identity (create SA, grant permissions)
3. Create disk. (make sure they are in the same zone with cluster)
4. Create a bucket (used by welder)
5. Download SAS image and upload to a safe place(I use same the project). 

``` 
helm install  --set serviceAccount.annotations.gcpServiceAccount="yonghao-test@terra-vdevel-glaring-bean-1388.iam.gserviceaccount.com" aou-sas-setup ./terra-app-setup-chart
```
```
helm install aou-sas-chart aou-sas-chart  --values aou-sas-chart/test_values.yaml 

```

### CODE REVIEW

**Review by Qi77Qi — APPROVED** [review]

**Comment by Qi77Qi** [comment]
things are working at this point...I'd like to merge this as is for now, but will be making improvements

**Comment by LizBaldo** [comment]
Up to you, for RStudio, we kept the PR open and published the chart from the feature branch and iterated on testing until things were working in leo. We ended up making a lot of little tweaks so it was nice to just do it within the PR.

**Review by jmthibault79 — APPROVED** [review]
Caveat: I have not attempted to use this, and I don't know much about Helm or GKE.  This is only a surface-level review.

File: terra-app-helm/aou-sas-chart/test_values.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+sasLicense: ""
+service:
+  type: ClusterIP
+  port: 80
+
+# TLS and rewrite rules provided by Leonardo
+ingress:
+  enabled: true
+  annotations:
+#   nginx.ingress.kubernetes.io/auth-tls-pass-certificate-to-upstream: "true"
```
[the remark this answers is not in the document]
Comment (Qi77Qi) [reply terra-app-helm/aou-sas-chart/test_values.yaml:26]: I'll delete them!

File: terra-app-helm/aou-sas-chart/test_values.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+   nginx.ingress.kubernetes.io/proxy-redirect-to: "$scheme://$host/sas/"
+
+# Provided by Leonardo
+nodeSelector: {}
+# cloud.google.com/gke-nodepool: "nodepool-name"
+
+# either GCE PD provided by Leonardo; or hostPath provided by terra-docker-local.sh
+persistence:
+  size: 100G
+  gcePersistentDisk: "gke-disk-test"
```
[the remark this answers is not in the document]
Comment (Qi77Qi) [reply terra-app-helm/aou-sas-chart/test_values.yaml:44]: hmm...so it's correct in a sense that in GCP, disks live under google compute I think...the name is used in other charts as well, so I'm gonna keep it as is for now

File: terra-app-helm/aou-sas-chart/test_values.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  - name: GOOGLE_PROJECT
+    value: "terra-vdevel-glaring-bean-1388"
+  - name: STAGING_BUCKET
+    value: "testbucket-sas-yonghao"
+  - name: CLUSTER_NAME
+    value: "all-of-us-123"
+  - name: RUNTIME_NAME
+    value:  "all-of-us-123"
+  - name: OWNER_EMAIL
+    value: "yonnghao@verily.com"
```
[the remark this answers is not in the document]
Comment (Qi77Qi) [reply terra-app-helm/aou-sas-chart/test_values.yaml:57]: I'm gonna update it to some dummy value

File: terra-app-helm/aou-sas-chart/values.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+license: ""
+
+# TLS and rewrite rules provided by Leonardo
+ingress:
+  enabled: true
+  path:
+    sas:
+    welder:
+  annotations:
+  # nginx.ingress.kubernetes.io/auth-tls-pass-certificate-to-upstream: "true"
```
[the remark this answers is not in the document]
Comment (Qi77Qi) [reply terra-app-helm/aou-sas-chart/values.yaml:34]: will update

File: terra-app-setup-chart/templates/service-account.yaml [hunk 5]
```diff
@@ -2,7 +2,7 @@
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: {{ .Values.serviceAccount.name | quote }}
```
[the remark this answers is not in the document]
Comment (Qi77Qi) [reply terra-app-setup-chart/templates/service-account.yaml:5]: good catch!...no...I'll revert it!

File: terra-app-helm/aou-sas-chart/test_values.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Qi77Qi) [reply terra-app-helm/aou-sas-chart/test_values.yaml:26]: I'll keep some of them cuz they might be useful once we start integrating with Leo for cert integration

**Pull request merged**